app/controllers/deploys_controller.rb

```diff
@@ -66,8 +66,11 @@ def create
       end
 
       format.json do
-        status = (@deploy.persisted? ? :created : :unprocessable_entity)
-        render json: @deploy.to_json, status: status, location: [current_project, @deploy]
+        if @deploy.persisted?
+          render json: @deploy.to_json, status: :created, location: [current_project, @deploy]
+        else
+          render_json_error 422, @deploy.errors
```

Comment:
rendering a `new` deploy does not help the requesting user much, breaks url/status/summary rendering and it does not show the errors either ... so changing to make it the same as the `rescue_from ActiveRecord::RecordInvalid`